flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -357,6 +362,28 @@ public void start() throws Exception {
 		return CompletableFuture.completedFuture(jobManagerServices.blobServer.getPort());
 	}
 
+	@Override
+	public CompletableFuture<Either<Throwable, SerializedJobExecutionResult>> getJobExecutionResult(
```

Comment:
I fear we cannot directly use a `Throwable` here, because it might be a user defined throwable. Therefore, we have to use `SerializedThrowable` instead.

Reply:
I'm actually wondering why we are differentiating between execution success and failure that much. Also in the failure case we should have a net runtime and accumulator values. The only difference is that in the failure case we have an additional `SerializedThrowable`.

We could for example have a common base class with a `JobExecutionResultSuccess` and `JobExecutionResultFailure` specialization (maybe also `JobExecutionResultCancelled` as a sub class of `JobExecutionResultFailure`). What do you think?

Reply:
Done.